### ISSUE TITLE
NullPointerException in TigerTreeVisualizer

### ISSUE THREAD

**Issue by arne-cl** [korpling/ANNIS#38]
Dear Thomas,

I imported a newer version of the PCC corpus (called "pcc176_tiger_mmax") into the Annis2.2.1 instance on the SFB server. When trying to open the tiger tree view (e.g. after searching for tiger.cat="NP"), I'll get this error:

```
HTTP Status 500 -
type Exception report
message

description The server encountered an internal error () that prevented it from fulfilling this request.

exception

java.lang.NullPointerException
	annis.frontend.servlets.visualizers.tree.TigerTreeVisualizer$DefaultLabeler.extractAnnotation(TigerTreeVisualizer.java:205)
	annis.frontend.servlets.visualizers.tree.TigerTreeVisualizer$DefaultLabeler.getLabel(TigerTreeVisualizer.java:190)
	annis.frontend.servlets.visualizers.tree.ConstituentLayouter.calculateNodePosition(ConstituentLayouter.java:394)
	annis.frontend.servlets.visualizers.tree.ConstituentLayouter.calculateNodePosition(ConstituentLayouter.java:370)
	annis.frontend.servlets.visualizers.tree.ConstituentLayouter.calculateNodePosition(ConstituentLayouter.java:370)
	annis.frontend.servlets.visualizers.tree.ConstituentLayouter.calculateNodePosition(ConstituentLayouter.java:370)
	annis.frontend.servlets.visualizers.tree.ConstituentLayouter.createLayout(ConstituentLayouter.java:351)
	annis.frontend.servlets.visualizers.tree.TigerTreeVisualizer.writeOutput(TigerTreeVisualizer.java:233)
	annis.frontend.servlets.VisualizerServlet.doGet(VisualizerServlet.java:197)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:617)
	javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	sun.reflect.GeneratedMethodAccessor24.invoke(Unknown Source)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:616)
	org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:270)
	org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:268)
	java.security.AccessController.doPrivileged(Native Method)
	javax.security.auth.Subject.doAsPrivileged(Subject.java:537)
	org.apache.catalina.security.SecurityUtil.execute(SecurityUtil.java:302)
	org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:163)
	java.security.AccessController.doPrivileged(Native Method)
	annis.frontend.filters.AuthenticationFilter.doFilter(AuthenticationFilter.java:54)
	sun.reflect.GeneratedMethodAccessor25.invoke(Unknown Source)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:616)
	org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:270)
	org.apache.catalina.security.SecurityUtil$1.run(SecurityUtil.java:268)
	java.security.AccessController.doPrivileged(Native Method)
	javax.security.auth.Subject.doAsPrivileged(Subject.java:537)
	org.apache.catalina.security.SecurityUtil.execute(SecurityUtil.java:302)
	org.apache.catalina.security.SecurityUtil.doAsPrivilege(SecurityUtil.java:243)
note The full stack trace of the root cause is available in the Apache Tomcat/6.0.32 logs.
```

I was able to reproduce the error with Annis 2.2.1 Kickstarter. If you'd like to look at the corpus, you'll find it on the SFB Annis server in the /home/annis/pcc176_tiger_mmax2_relANNIS folder.

Best regards,
Arne

**Comment by thomaskrause** [comment]
This should be fixed in a54b2ac9ed72ee07a7affdb486b62af7706ee9bf or earlier

**Comment by arne-cl** [comment]
What a blast from the past!

Best,
Arne
